From 1068d7523f6a0bf2293f5af893b49ed0d08ef865 Mon Sep 17 00:00:00 2001
From: Julien STEPHAN <jstephan@baylibre.com>
Date: Mon, 28 Dec 2020 10:29:37 +0000
Subject: [PATCH] don't use __system_properties

because we don't compile on android, `__system_property_find` is
undefined, so use a `#if defined(__BIONIC__)` preprocessor to protect it
and always return the default value for `__system_property_find`

Signed-off-by: Julien STEPHAN <jstephan@baylibre.com>
---
 SystemPropertiesUtils.hpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/SystemPropertiesUtils.hpp b/SystemPropertiesUtils.hpp
index 78e5ac7..cbb57c0 100644
--- a/SystemPropertiesUtils.hpp
+++ b/SystemPropertiesUtils.hpp
@@ -8,7 +8,9 @@
 #include <stdio.h>
 #include <string>
 #include <iostream>
+#if defined(__BIONIC__)
 #include <sys/system_properties.h>
+#endif
 #include <log/log.h>
 
 namespace {
@@ -46,6 +48,7 @@ void GetCapabilitiesProperties([[maybe_unused]]void* cookie,
 template<typename T>
 T ParseSystemProperty(const char* name, T defaultValue)
 {
+#if defined(__BIONIC__)
     try
     {
         const prop_info *pInfo = __system_property_find(name);
@@ -80,5 +83,8 @@ T ParseSystemProperty(const char* name, T defaultValue)
     messageBuilder << "ArmnnDriver::ParseSystemProperty(): Falling back to default value [" << defaultValue << "]";
     ALOGD("%s", messageBuilder.str().c_str());
     return defaultValue;
+#else
+    return defaultValue;
+#endif
 }
 } //namespace
